### PR TITLE
feat(jsx/precompile): Normalization and stringification of attribute values as `renderToString`

### LINKED ISSUE #3427: `style=` objects do not work with `jsx: precompile` on Deno
### What version of Hono are you using?

4.5.9

### What runtime/platform is your app running on?

Deno 1.45.5

### What steps can reproduce the bug?

When using the `"precompile"` JSX option in Deno, passing an object to the `style` attribute does not work correctly. Instead, it always outputs `[object Object]`.

```jsx
return c.render(
  <div style={{ display: 'none' }} />
);
```

### What is the expected behavior?

With `"jsx": "react-jsx"`:

```html
<div style="display:none"></div>
```

### What do you see instead?

With `"jsx": "precompile"`:

```html
<div style="[object Object]"></div>
```

### Additional information

_No response_

### PR DESCRIPTION
**Pull request by usualoma**
fix #3427

The “precompile” process converts attribute values in the same way as the normal `renderToString` process.
https://github.com/usualoma/hono/blob/fd80df6a0f03876cf6ac2b12b4b7b812c19ca616/src/jsx/base.ts#L180-L227

### The author should do the following, if applicable

- [x] Add tests
- [x] Run tests
- [x] `bun run format:fix && bun run lint:fix` to format the code

### CODE REVIEW

**Comment by usualoma** [comment]
It'll need some tweaking.

**Comment by usualoma** [comment]
@yusukebe 
Would you please review?

File: src/jsx/jsx-runtime.ts [hunk 1]
```diff
@@ -6,13 +6,45 @@
...
 export { html as jsxTemplate }
+
 export const jsxAttr = (
-  name: string,
-  value: string | Promise<string>
-): HtmlEscapedString | Promise<HtmlEscapedString> =>
-  typeof value === 'string' ? raw(name + '="' + html`${value}` + '"') : html`${name}="${value}"`
+  key: string,
+  v: string | Promise<string> | Record<string, string | number | null | undefined | boolean>
+): HtmlEscapedString | Promise<HtmlEscapedString> => {
```
Comment (yusukebe) [inline src/jsx/jsx-runtime.ts:20]: Can we create one common function for these lines and https://github.com/usualoma/hono/blob/fd80df6a0f03876cf6ac2b12b4b7b812c19ca616/src/jsx/base.ts#L180-L227 and use it from the both?

Like this, though, I think `objectToAttribute` is not the best name.

```ts
export const objectToAttribute = (key: string, v: any, buffer: any) => {
  if (v === null || v === undefined) {
    return
  }
  if (key === 'style' && typeof v === 'object') {
    // object to style strings
    let styleStr = ''
    styleObjectForEach(v as Record<string, string | number>, (property, value) => {
      if (value != null) {
        styleStr += `${styleStr ? ';' : ''}${property}:${value}`
      }
    })
    escapeToBuffer(styleStr, buffer)
    buffer[0] += '"'
  } else if (typeof v === 'boolean' && booleanAttributes.includes(key)) {
    return buffer
  } else if (typeof v === 'string') {
  // ...
  return buffer
}
```

File: src/jsx/jsx-runtime.ts [hunk 2]
```diff
@@ -6,13 +6,45 @@
 export { jsxDEV as jsx, Fragment } from './jsx-dev-runtime'
 export { jsxDEV as jsxs } from './jsx-dev-runtime'
 export type { JSX } from './jsx-dev-runtime'
-
 import { html, raw } from '../helper/html'
-import type { HtmlEscapedString } from '../utils/html'
+import type { HtmlEscapedString, StringBuffer, HtmlEscaped } from '../utils/html'
+import { escapeToBuffer, stringBufferToString } from '../utils/html'
+import { booleanAttributes } from './base'
```
Comment (yusukebe) [inline src/jsx/jsx-runtime.ts:12]: This `booleanAttributes` is never used. Could you remove this line?

**Comment by yusukebe** [comment]
Hey @usualoma 

Thank you for the PR. I've left comments!

File: src/jsx/jsx-runtime.ts (hunk 2 above)
[the remark this answers is not in the document]
Comment (usualoma) [reply src/jsx/jsx-runtime.ts]: fixed b337b23

**Comment by usualoma** [comment]
Key normalization is done by deno, so processing by hono was removed.  4405e69

File: src/jsx/jsx-runtime.ts (hunk 1 above)
Comment (usualoma) [reply src/jsx/jsx-runtime.ts:19]: @yusukebe 
I also considered making it common, but if it is made common, there will inevitably be overhead for both sides in the following areas.

https://github.com/usualoma/hono/blob/4405e69ae8ab419b2dab36dfd25c605a9c50fca3/src/jsx/base.ts#L202
https://github.com/usualoma/hono/blob/4405e69ae8ab419b2dab36dfd25c605a9c50fca3/src/jsx/base.ts#L205

I don't think it's a big overhead, but I want to be particularly conscious of performance when stringifying JSX, so I want to keep it separate for each implementation for optimization. What do you think?

Comment (yusukebe) [reply src/jsx/jsx-runtime.ts:19]: @usualoma 

Understood well. Let's go with this. Thank you for spending for considering it!

**Review by yusukebe — APPROVED** [review]
LGTM!

**Comment by yusukebe** [comment]
@usualoma 

Looks good! Thank you.

**Pull request merged**